src/components/Calculator.js

```diff
@@ -1,55 +1,45 @@
-  };
-
-  showResult = () => {
-    const { total, next, operation } = this.state;
-    const show = (total || '') + (operation || '') + (next || '');
-
-    return (show || '0');
+const Calculator = () => {
+  const [state, setState] = useState({});
+  const handleClick = (ev) => {
+    setState({ ...state, ...calculate(state, ev.target.name) });
   };
   
-  render() {
-    return (
-      <div className="calculator">
-        <div className="number">
-          <p>{this.showResult()}</p>
-        </div>
-        <section className="buttons">
-          <button type="button" name="AC" onClick={(ev) => this.handleClick(ev.target.name)}>AC</button>
-          <button type="button" name="+/-" onClick={(ev) => this.handleClick(ev.target.name)}>+/-</button>
-          <button type="button" name="%" onClick={(ev) => this.handleClick(ev.target.name)}>%</button>
-          <button type="button" name="&#247;" className="orange" onClick={(ev) => this.handleClick(ev.target.name)}>&#247;</button>
-          <button type="button" name="7" onClick={(ev) => this.handleClick(ev.target.name)}>7</button>
-          <button type="button" name="8" onClick={(ev) => this.handleClick(ev.target.name)}>8</button>
-          <button type="button" name="9" onClick={(ev) => this.handleClick(ev.target.name)}>9</button>
-          <button type="button" name="x" className="orange" onClick={(ev) => this.handleClick(ev.target.name)}>x</button>
-          <button type="button" name="4" onClick={(ev) => this.handleClick(ev.target.name)}>4</button>
-          <button type="button" name="5" onClick={(ev) => this.handleClick(ev.target.name)}>5</button>
-          <button type="button" name="6" onClick={(ev) => this.handleClick(ev.target.name)}>6</button>
-          <button type="button" name="-" className="orange" onClick={(ev) => this.handleClick(ev.target.name)}>-</button>
-          <button type="button" name="1" onClick={(ev) => this.handleClick(ev.target.name)}>1</button>
-          <button type="button" name="2" onClick={(ev) => this.handleClick(ev.target.name)}>2</button>
-          <button type="button" name="3" onClick={(ev) => this.handleClick(ev.target.name)}>3</button>
-          <button type="button" name="+" className="orange" onClick={(ev) => this.handleClick(ev.target.name)}>+</button>
-          <button type="button" name="0" className="zero" onClick={(ev) => this.handleClick(ev.target.name)}>0</button>
-          <button type="button" name="." onClick={(ev) => this.handleClick(ev.target.name)}>.</button>
-          <button type="button" name="=" className="orange" onClick={(ev) => this.handleClick(ev.target.name)}>=</button>
-        </section>
+  const { total, next, operation } = state;
+  const showResult = (total || '') + (operation || '') + (next || '');
+  
+  return (
+    <div className="calculator">
+      <div className="number">
+        <p>{showResult || '0'}</p>
       </div>
-    );
-  }
-}
+      <section className="buttons">
```

Comment:
- [ ] **[Optional] Please consider refactoring this code so that your code follows the DRY approach.**

Each line of code is creating the same component. You can create a collection of data to be passed to each button component and then you can iterate through that collection.

Reply:
Thank you for this note. I will fix it promptly! 👍